code/game/gamemodes/miniantags/tourist/tourist_arrivals.dm

```diff
@@ -0,0 +1,156 @@
+		var/raffle_name = pick("Galactic Getaway Raffle", "Cosmic Jackpot Raffle", "Nebula Nonsense Raffle", "Greytide Giveaway Raffle", "Toolbox Treasure Raffle")
+		GLOB.minor_announcement.Announce("The lucky winners of the Nanotrasen raffle, 'Nanotrasen [raffle_name],' are arriving at [station_name()] shortly. Please welcome them warmly, they'll be staying with you until the end of your shift!")
+// Greets the player, announces objectives!
+/datum/event/tourist_arrivals/proc/greeting(mob/living/carbon/human/M)
+	var/list/greeting = list()
+	greeting.Add("<span class='boldnotice'><font size=3>You are a tourist!</font></span>")
+	greeting.Add("<b>You were chosen as a lucky winner of Nanotrasen's exclusive raffle! Winning a visit to a nearby Nanotrasen Research Station!</b>")
+	greeting.Add("<b>Enjoy your exclusive tour and make the most of your time exploring our state-of-the-art facilities!</b>")
+	greeting.Add("<span class='notice'><br>Your current objectives are:</span>")
+	greeting.Add(M.mind.prepare_announce_objectives(FALSE))
+	to_chat(M, chat_box_green(greeting.Join("<br>")))
+
+// All of this code down here is from ert.dm.
+/datum/event/tourist_arrivals/proc/set_appearance(mob/living/carbon/human/M)
+
+	var/obj/item/organ/external/head/head_organ = M.get_organ("head")
+	var/hair_c = pick("#8B4513", "#000000", "#FF4500", "#FFD700") // Brown, black, red, blonde
+	var/eye_c = pick("#000000", "#8B4513", "1E90FF") // Black, brown, blue
+	var/skin_tone = rand(-120, 20) // A range of skin colors (This doesn't work, result is always pale white)
+
+	head_organ.facial_colour = hair_c
+	head_organ.sec_facial_colour = hair_c
+	head_organ.hair_colour = hair_c
+	head_organ.sec_hair_colour = hair_c
+	M.change_eye_color(eye_c)
+	M.s_tone = skin_tone
+	head_organ.h_style = random_hair_style(M.gender, head_organ.dna.species.name)
+	head_organ.f_style = random_facial_hair_style(M.gender, head_organ.dna.species.name)
+
+	M.regenerate_icons()
+	M.update_body()
+	M.update_dna()
+
+// Tourist datum stuff, mostly being species and outfit.
+/datum/tourist
+	var/tourist_outfit
+	var/tourist_species
+
+/datum/tourist/human
+	tourist_outfit = /datum/outfit/admin/tourist
+	tourist_species = /mob/living/carbon/human
+
+/datum/tourist/unathi
+	tourist_species = /mob/living/carbon/human/unathi
+	tourist_outfit = /datum/outfit/admin/tourist
+
+/datum/tourist/vulp
+	tourist_species = /mob/living/carbon/human/vulpkanin
+	tourist_outfit = /datum/outfit/admin/tourist
+/datum/tourist/ipc
+	tourist_species = /mob/living/carbon/human/machine
+	tourist_outfit = /datum/outfit/admin/tourist
+
+/datum/tourist/skrell
+	tourist_species = /mob/living/carbon/human/skrell
+	tourist_outfit = /datum/outfit/admin/tourist
+
+/datum/tourist/grey
+	tourist_species = /mob/living/carbon/human/grey
+	tourist_outfit = /datum/outfit/admin/tourist
+
+/datum/tourist/nian
+	tourist_species = /mob/living/carbon/human/moth
+	tourist_outfit = /datum/outfit/admin/tourist
```

Comment:
please clean up the spacing here and ensure that the variable definitions follow consistent ordering